### ISSUE TITLE
`os.path.join()` on Windows giving wrong path format for posix systems when calling run_submission().

### ISSUE THREAD

**Issue by saltball** [deepmodeling/dpdispatcher#78]
# Some environment info:
dpdispatcher version: 0.3.12 via `pip`
python: 3.8.10
# Description
I'm not sure if `dpdispatcher` is only for linux or working on one same platform.
When I try to call my `.py` script on Windows and `ssh` to some linux platform (for example `wsl`), with `Machine` configuration as:
```python
machine = Machine(batch_type="Shell",
                  context_type="SSHContext",
                  local_root="windowspath/",
                  remote_root="/home/some/ubuntu/path/",
                  remote_profile={
                  ...
                  }
                  )
```
I always got a error code 2 in calling `tar xf ....`.
# Locate the problem
It seems line 445 in `ssh_context.py` where using `os.path.join()` show different result on Windows and Linux.
When I run my script on Windows, dpdispatcher will upload task to a `.tgz` on Ubuntu named weirdly `'xxxx\xxxx'` (and in `remote_root` directory) but not the file `xxxx` in the directory `xxxx`.
And `\` is only used in path of Windows but not Linux (or any other posix system).
# Proposal 
If necessary, I suggest using library `pathlib` to fix this.
**An example**: 
(maybe work for both Windows and other posix system, because Python on Windows can correctly get the path in posix format.) (**Not tested on other situations**)
(change line 446 in `ssh_context.py` to)
```python
to_f=pathlib.PurePath(os.path.join(self.remote_root, of)).as_posix()
```
And there are other codes should be modified. Thus I wander if `dpdispatcher` is only for linux or working on one same platform. If so, maybe doc should be modified to remind users.

**Comment by njzjz** [comment]
Could you submit a pull request to fix it? Both @felix5572 and I do not use Windows for work, so it's hard to debug,

**Comment by saltball** [comment]
@njzjz @felix5572 Please check PR #81, let me know if it would cause any new problem.

**Comment by saltball** [comment]
I think path issue has been fixed with #81.
